docs_src/manuals/user_manual/execution_model.hrst

```diff
@@ -3,35 +3,31 @@
 iteration space, usually referred to as `I` and `J` dimensions identify the horizontal dimension. There is no
-prescription on in what order stencil operators in different points of the same `IJ` plane will be executed. Stencil
-operators in the third dimension of the iteration space, usually referred as `K` or vertical dimension, can have
-prescribed order of executions. There are three different ways of execution policies for the `K` dimension:
+prescription for the order of stencil operators in different points of the same `IJ` plane. Stencil operators in the
+third dimension of the iteration space, usually referred as `K` or vertical dimension, can have prescribed order
+of executions. There are three different execution policies for the `K` dimension:
 
 - `forward`: The computation at index `k` in the vertical dimension is executed after index `k - 1` for all points in
   the horizontal plane;
 - `backward`: The computation at index `k` in the vertical dimension is executed after index `k + 1` for all points in
   the horizontal plane;
 - `parallel`: No order is specified and execution can happen concurrently.
 
-More concretely, a multistage is a list of stages (implemented with stencil operators) to be executed with a certain
-execution policy.  A computation combines several multistages, and will execute one multistage after the other.
+More concretely, a multistage is a list of stages, implemented with stencil operators, to be executed with a certain
+execution policy. A computation combines several multistages and executes one multistage after the other.
 
-- For each `IJ` plane, the stages of a multistage will be executed strictly one after the other. This means, that a
-  stage can assume, that the previous stage has been applied to the the whole `IJ` plane before it is executed. The user
-  can explicitly create independent stages, that don't require this restriction.
+- For each `IJ` plane, the stages of a multistage will be executed strictly one after the other. This means that a
+  stage can assume that the previous stage has been applied to the the whole `IJ` plane before its execution. The user
+  can explicitly create independent stages that do not require this restriction.
 - If the execution policy is `parallel`, a stage cannot impose any assumptions on which stages are applied before in
   another `IJ` plane.
-- If the execution policy is `forward`, it is guaranteed, that if a stage is executed at index ``k``,
-  then all previous stages of
-  the multistage were already applied to the same column with smaller ``k``. There is no guarantee that previous stages
-  of the multistage are
+- If the execution policy is `forward`, it is guaranteed that if a stage is executed at index ``k`` then all stages of
+  the multistage were already applied to the same column with smaller ``k``, and that no stages of the multistage are
```

Comment:
This is wrong, depending on the backend (simple example: the naive one), previous stages of the multistage _are_ already applied on the same column with larger k.

Reply:
Good catch. Will fix now.